### PR TITLE
feat!: refactor in TS

### PR DESCRIPTION
**Pull request by pom421**
I use this module to test my TS rewriting and publishing skills ;). 

@revolunet tell me if you want this PR, since it adds a bit of tooling complexity.

It causes some breaking changes as well, since I didn't manage to have a default export but it works with a named export.

Also, I tried to stick with the Python version, which has changed a bit the signature (zone property in the options object). 
They added the support of "Jour de l'abolition de l'esclavage", which can be useful in this version.

We could add it in another PR. 

Also : 
- add prettier/eslint config
- add vscode support with autoformating
- add a license file

### CODE REVIEW

**Comment by revolunet** [comment]
il faudra ajouter `BREAKING CHANGE:` dans le footer du commit message pour déclencher une release majeure

**Comment by pom421** [comment]
@revolunet il faudrait ajouter le run des tests dans l'action, pour s'assurer que c'est ok.

**Comment by revolunet** [comment]
les tests sont bien runnés dans l'action de build/release

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "editor.codeActionsOnSave": {
+        "source.fixAll.eslint": true
+    },
+    "editor.formatOnSave": false,
+    "workbench.colorCustomizations": {
+        "statusBar.background": "#fd935698" // Change this color!
+    },
+    // https://github.com/colinhacks/zod/issues/443
```
Comment (revolunet) [inline .vscode/settings.json:9]: comments in JSON ?

**Review by revolunet — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (pom421) [reply .vscode/settings.json:9]: Not a problem for vscode

**Pull request merged**